pkg/controller/petset/pet_set.go

```diff
@@ -226,12 +227,41 @@ func (psc *StatefulSetController) getPodsForStatefulSet(ps *apps.StatefulSet) ([
 	}
 	// TODO: Do we need to copy?
 	result := make([]*api.Pod, 0, len(pods))
+	// Check parent statefulset annotation to avoid fighting with other controllers, it's safer than using name prefix
 	for i := range pods {
-		result = append(result, &(*pods[i]))
+		if ps.UID == getStatefulSetUID(pods[i]) {
+			result = append(result, &(*pods[i]))
+		}
 	}
 	return result, nil
 }
 
+// getStatefulSetUIDFromPod extracts UID of pod's parent statefulset; return an empty UID if not found
+func getStatefulSetUID(pod *api.Pod) types.UID {
+	creatorRefJson, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation]
+	if !found {
+		glog.V(4).Infof("Unable to get parent UID: pod %s/%s with no created-by annotation", pod.Namespace, pod.Name)
+		return types.UID("")
+	}
+	var sr api.SerializedReference
+	err := json.Unmarshal([]byte(creatorRefJson), &sr)
+	if err != nil {
+		glog.V(4).Infof("Unable to get parent UID: pod %s/%s with unparsable created-by annotation: %v", pod.Namespace, pod.Name, err)
+		return types.UID("")
+	}
+	if sr.Reference.Kind != "StatefulSet" {
+		glog.V(4).Infof("Unable to get parent UID: pod %s/%s with non-StatefulSet parent", pod.Namespace, pod.Name)
+		return types.UID("")
+	}
+	// Don't believe a pod that claims to have a parent in a different namespace.
+	if sr.Reference.Namespace != pod.Namespace {
```

Comment:
Will it? I tried creating a StatefulSet in current namespace (say, "default"). It created pods with the created-by annotation with "namespace: default"